types/datum.go

```diff
@@ -1019,8 +1019,18 @@ func (d *Datum) convertToMysqlDuration(sc *stmtctx.StatementContext, target *Fie
 		timeStr, err := d.ToString()
 		if err != nil {
 			return ret, errors.Trace(err)
-		} else if timeStr[0] == '-' {
-			return ret, ErrInvalidTimeFormat.Gen("Incorrect time value '%s'", timeStr)
+		}
+		timeNum, err := d.ToInt64(sc)
+		if err != nil {
+			return ret, errors.Trace(err)
+		}
+		if timeNum > MaxDuration && timeNum < 10000000000 {
```

Comment:
what if timeNum > 10000000000

Reply:
timeNum > MaxDuration and < 10000000000 must be a wrong value..
but timeNum > 10000000000 will try treat as `'YYYYMMDDHHmmss'`

specially timeNum < 0 and timeNum < -10000000000 should NOT treat as `'YYYYMMDDHHmmss'`

https://github.com/mysql/mysql-server/blob/e7586bb9c11953a81c816ca10708f612403edc24/sql-common/my_time.c#L846